### PR TITLE
Apply some @m-kuhn comments

### PR DESCRIPTION
**Pull request by sbrunner**
on the commit bc1ad2ce82196402ffa83118b7fd3c2f195a4414

### CODE REVIEW

File: src/providers/wfs/qgswfsprovider.h [hunk 1]
```diff
@@ -137,7 +137,22 @@ class QgsWFSProvider : public QgsVectorDataProvider
      * Get some metadata description of the provider.
      * @return The provider metadata
      */
-    virtual QgsStringMap metadata();
+    virtual QVariantMap metadata();
```
Comment (m-kuhn) [inline src/providers/wfs/qgswfsprovider.h]: Shouldn't these methods override the one in the base class?

    virtual QVariantMap metadata() const override;

**Review by m-kuhn — CHANGES_REQUESTED** [review]

**Comment by m-kuhn** [comment]
Thanks, that looks a lot better to me!

File: python/core/qgsvectordataprovider.sip [hunk 2]
```diff
@@ -401,7 +401,22 @@ class QgsVectorDataProvider : QgsDataProvider
...
      */
-    virtual QMap<QString, QString> metadata() const;
+    virtual QVariantMap metadata() const;
+
+    /**
+     * Get the translated metadata key.
+     * @param md_key The metadata key
+     * @return The translated metadata value
+     */
+    virtual QString translatedMetadataKey( QString md_key ) const;
```
Comment (m-kuhn) [inline python/core/qgsvectordataprovider.sip]: * Imperatively call it translateMetadataKey() and translateMetadataValue() (they are more transformers than getters)
* pass a `const QString& mdKey ` as parameter

**Review by m-kuhn — CHANGES_REQUESTED** [review]

File: python/core/qgsvectordataprovider.sip (hunk 2 above)
Comment (sbrunner) [inline python/core/qgsvectordataprovider.sip]: +1

**Comment by sbrunner** [comment]
Thanks @m-kuhn :-)

**Pull request merged**